src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+import { postNewComment } from "../utils/api";
+import { UserContext } from "../contexts/User";
+import { useContext } from "react";
+
+const CommentAdder = ({ review_id, setComments }) => {
+  const { user } = useContext(UserContext);
+  const [newComment, setNewComment] = useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    postNewComment(review_id, user.username, newComment).then(({ comment }) => {
+      setComments((currComments) => {
+        return [comment, ...currComments];
+      });
+      setNewComment("");
+    });
+  };
+
+  return (
+    <form className="comment__adder" onSubmit={(e) => handleSubmit(e)}>
```

Comment:
You don't actually need to pass the event here, this could be re-written as `onSubmit={handleSubmit}`, the handleSubmit will be invoked with e (the event) when the form is submitted anyway!